libs/vkd3d/command.c

```diff
@@ -8550,6 +8543,25 @@ static const struct vkd3d_format *vkd3d_clear_uav_find_uint_format(struct d3d12_
     return vkd3d_get_format(device, uint_format, false);
 }
 
+static bool vkd3d_clear_uav_check_uint_format_compatibility(struct d3d12_device *device, const struct vkd3d_format *resource_format, const struct vkd3d_format *uint_format)
+{
+    const struct vkd3d_format_compatibility_list *compat;
+    unsigned int i;
+
+    if (resource_format->vk_format == uint_format->vk_format)
+        return true;
+
+    compat = &device->format_compatibility_lists[resource_format->dxgi_format];
+
+    for (i = 0; i < compat->format_count; i++)
+    {
+        if (compat->vk_formats[i] == uint_format->vk_format)
+            return true;
```

Comment:
I assume TYPELESS will always pass this check?

Reply:
Yes, all typeless formats have a corresponding UINT format.